config/routes.rb

```diff
@@ -1,7 +1,12 @@
 Rails.application.routes.draw do
+
+  if Settings.google_verification
+    match "/#{Settings.google_verification}.html", to: proc { |env| [200, {}, ["google-site-verification: #{Settings.google_verification}.html"]] }, via: :get
+  end
+
   ActiveAdmin.routes(self)
-  # We remove the sign_up path name so as not to allow users to sign in with username and password.
-  devise_for :users, :controllers => { :omniauth_callbacks => "omniauth_callbacks" }, path_names: { sign_up: ''}
```

Comment:
Killed `path_names` because it's misleading - it doesn't work. The real fix exists in the user model file.